lib/elixir/lib/calendar.ex

```diff
@@ -1038,16 +1038,13 @@ defmodule Time do
     diff_parts = parts1 * ppd2 - parts2 * ppd1
 
     # Keep integers in day fraction low.
-    gcd = gcd(diff_parts, diff_ppd)
+    gcd = Integer.gcd(diff_parts, diff_ppd)
     diff_parts = div(diff_parts, gcd)
     diff_ppd = div(diff_ppd, gcd)
 
-    microseconds =
-      {diff_parts, diff_ppd}
-      |> Calendar.ISO.time_from_day_fraction
-      |> to_microsecond
-
-    System.convert_time_unit(microseconds, unit, :microsecond)
+    {0, {diff_parts, diff_ppd}}
+    |> Calendar.ISO.rata_die_to_microseconds
+    |> System.convert_time_unit(:microseconds, unit)
```

Comment:
Maybe we should move this call to `rata_die_to_microseconds` and call it `rata_die_to_unit`?